src/clj/puppetlabs/services/versioned_code_service/versioned_code_core.clj

```diff
@@ -55,13 +59,16 @@
             (string/trim-newline stdout))
           (do
             (log/error (nonzero-msg code-id-script exit-code stdout stderr))
-            nil)))
+            (throw-execution-error!))))
       (catch IllegalArgumentException e
-        (log-execution-error! e))
+        (log-execution-error! e)
+        (throw-execution-error!))
       (catch IOException e
-        (log-execution-error! e))
+        (log-execution-error! e)
+        (throw-execution-error!))
       (catch InterruptedException e
-        (log-execution-error! e)))))
+        (log-execution-error! e)
+        (throw-execution-error!)))))
```

Comment:
This looks like it's turned into a bit of a mess.

First of all, why are we catching `IllegalArgumentException` here?  That looks a little suspicious; I doubt that's to catch an error from a call into 3rd party code, and if it's our code that we're expecting to throw an IAE, we should change that to use a slingshot exception if possible.

Second, instead of logging the original exception and then throwing a different one, can we:
a) Just remove this exception handling altogether?
b) Failing that, remove the logging and "wrap and re-throw" i.e. ...
```clj
(catch IOException e
  (throw (IllegalStateException. code-id-error-msg e))
```
(note that the original exception `e` is "wrapped" into the 2nd exception).